### PR TITLE
ci: lint cargo.lock in CI

### PR DESCRIPTION
**Pull request by elliotwutingfeng**
## Current Behavior

When a PR updates `Cargo.lock` file, we should ensure that the lock file was generated by `cargo command` and not tampered with. 

## Proposed Changes

Add the following jobs to CI `rust.yml` workflow:

   - Gets the current SHA of Cargo.lock generated by PR/user.
   - Deleting the current Cargo.lock
   - Generating a new Cargo.lock, possibly calling cargo check command.
   - Getting the SHA of the new lock file
   - Ensure the SHA generated by user PR is same as that of CI

Addresses #3572

## Checks

- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [x] All commits in this Pull Request follow the Ockam [commit message convention](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages).
- [x] I accept the Ockam Community [Code of Conduct](https://github.com/build-trust/.github/blob/main/CODE_OF_CONDUCT.md).
- [x] I have accepted the Ockam [Contributor License Agreement](https://github.com/build-trust/ockam-contributors/blob/main/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam-contributors](https://github.com/build-trust/ockam-contributors) repository.

### CODE REVIEW

File: .github/workflows/rust.yml
Comment (metaclips) [inline .github/workflows/rust.yml]: Let's also cache cargo build so that `cargo check` command will be fast, example [here](https://github.com/build-trust/ockam/blob/d792ddcec1194c0d666ed69c5bb7bf5409470562/.github/workflows/rust.yml#L89-L93)

**Review by metaclips — CHANGES_REQUESTED** [review]
Thanks for an amazing PR!

File: .github/workflows/rust.yml
Comment (metaclips) [reply .github/workflows/rust.yml]: @elliotwutingfeng seems our CI isn't happy with the current commit message as it doesn't allow uppercase, can you instead rename the commit to
ci: lint cargo.lock in \`CI\`
or
ci: lint cargo.lock in ci
or whichever amazing and descriptive commit message you can think of 🙂

**Comment by mrinalwadhwa** [comment]
@elliotwutingfeng Thank you for spending time on this, before we can merge please

- [x] Accept the Ockam [Contributor License Agreement](https://github.com/build-trust/ockam-contributors/blob/main/CLA.md) by adding your Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam-contributors](https://github.com/build-trust/ockam-contributors) repository.

The easiest way to do this is to [edit the CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/edit/main/CONTRIBUTORS.csv) file in the github web UI and create a PR, this will make the commit as verified.

**Comment by elliotwutingfeng** [comment]
Replaced `sha512sum` with `sha384sum` as `sha512sum` is vulnerable to [length-extension attacks](https://pycryptodome.readthedocs.io/en/latest/src/hash/sha512.html#sha-512-sha-512-224-sha-512-256) while `sha384sum` is [not](https://pycryptodome.readthedocs.io/en/latest/src/hash/sha384.html#sha-384).

**Comment by mrinalwadhwa** [comment]
@elliotwutingfeng @metaclips I've been thinking about this PR. I think we should run this check only if a PR changes Cargo.lock so we're not forcing contributors to update cargo locks before sending unrelated PRs. What do you think?

**Comment by elliotwutingfeng** [comment]
@mrinalwadhwa Yes, I think that makes sense.

File: .github/workflows/rust.yml
Comment (mrinalwadhwa) [inline .github/workflows/rust.yml]: @elliotwutingfeng Instead of using an extra dependency. Would be better to just add another workflow, since github can trigger only on specific path changes:

https://github.com/build-trust/ockam/blob/develop/.github/workflows/rust.yml#L12-L32

**Review by mrinalwadhwa — CHANGES_REQUESTED** [review]
left a suggestion

File: .github/workflows/rust.yml
Comment (elliotwutingfeng) [reply .github/workflows/rust.yml]: I set it to check for `**.toml` as well since changing it would usually change `Cargo.lock`.

**Review by metaclips — DISMISSED** [review]
Approved 🚀. Thanks @elliotwutingfeng for taking this through. Can you squash your commits so that we can merge right away.

**Comment by elliotwutingfeng** [comment]
Done!

**Review by mrinalwadhwa — APPROVED** [review]
@elliotwutingfeng Thank you for an awesome first contribution 🥳 🎈

**Pull request merged**
